pkg/kvstore/etcd.go

```diff
@@ -63,6 +71,34 @@ type EtcdLocker struct {
 	path      string
 }
 
+func checkMinVersionOfEndpoint(etcdEP string) error {
+	r, err := http.Get(etcdEP + "/version")
+	if err != nil {
+		return fmt.Errorf("error contacting etcd endpoint %q: %s", etcdEP, err)
+	}
+	if r.StatusCode != 200 {
+		return fmt.Errorf("error contacting etcd endpoint %q: %s", etcdEP, err)
+	}
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("error reading request from etcd endpoint %q: %s", etcdEP, err)
+	}
+	r.Body.Close()
+	var ver etcdVersion.Versions
+	if err := json.Unmarshal(b, &ver); err != nil {
+		return fmt.Errorf("error unmarshalling etcd cluster version: %s", err)
+	}
+	v, err := version.NewVersion(ver.Cluster)
+	if err != nil {
+		return fmt.Errorf("error creating a new version object for etcd cluster version: %s", err)
+	}
+	if !minEVersion.Check(v) {
+		log.Fatalf("ETCD cluster doesn't meet the minimal requirement, please upgrade it to %s", minEVersion.String())
```

Comment:
I don't think that you should `log.Fatalf` here, instead this should be handled in the caller.